lib/mkl/linalg.jl

```diff
@@ -49,6 +49,11 @@ function gemm_dispatch!(C::oneStridedVecOrMat, A, B, alpha::Number=true, beta::N
     end
 end
 
+function LinearAlgebra.axpy!(alpha::Number, x::oneStridedVecOrMat{<:onemklFloat}, y::oneStridedVecOrMat{<:onemklFloat}) where T<:Union{Float16, ComplexF16, onemklFloat}
```

Comment:
We currently do not codes to wrap Float16, ComplexF16 to call onemkl::apxy, so please remove the Float16, ComplexF16 from this.

A longer term project is the enable Float16 for all supported MKL functions.
MKL does support Float16, BFloat16
https://spec.oneapi.io/versions/latest/elements/oneMKL/source/domains/blas/axpy.html.

If we can find a way to write C wrapper for them, that will be great to hook them up.  One question is how to specify float16 in C.  Clang compiler supports __fp16 so we may be able to use it.  Please also investigate how it is supported in CUDA.jl and AMDGPU.jl.

Reply:
Good catch. yes, half types should be removed from union. I will investigate half type of implementation in CUDA.jl and AMDGPU.jl.